nix/nixos/cardano-node-service.nix

```
@@ -266,10 +275,19 @@ in {
     } // optionalAttrs (! cfg.instanced) {
       wantedBy = [ "multi-user.target" ];
     };
-    assertions = [{
-      assertion = lib.hasPrefix stateDirBase cfg.stateDir;
-      message =
-        "The option services.cardano-node.stateDir should have ${stateDirBase} as a prefix!";
-    }];
+    assertions = [
+      {
+        assertion = lib.hasPrefix stateDirBase cfg.stateDir;
+        message = "The option services.cardano-node.stateDir should have ${stateDirBase} as a prefix!";
+      }
+      {
+        assertion = (cfg.genesisHash == null) != (cfg.genesisHashPath == null);
```

That assertion (XOR) also cover the case of the next one. So the next assertion can be removed (and the message of this one complexified) or change this assertion to not cover the case of next one:
```suggestion
        assertion = !((cfg.genesisHash != null) && (cfg.genesisHashPath != null));
```